lib/pubnub/client.rb

```diff
@@ -379,10 +379,7 @@ def setup_app(options)
       Pubnub.logger = options[:logger] || Logger.new('pubnub.log')
       Concurrent.global_logger = Pubnub.logger
       @subscriber = Subscriber.new(self)
-      user_id = options[:user_id]
-      if user_id != nil
-        options[:uuid] = user_id
-      end
+      options[:uuid] = options[:user_id] unless options[:user_id].nil?
```

Comment:
@kleewho wondering why this way, and not in suggested? 
```ruby
options[:user_id] = options[:uuid] unless options[:uuid].nil?
```

For user we have getter / setter, but it will be better to use internally proper keys already.

Reply:
> > The question is do our users use `.env[:uuid]`? If yes then I don't think we should make this change in this way yet. We should first deprecate uuid to let them some time to adjust and only then drop uuid.
> 
> `.env[:uuid]` is internally used had (suggested search version is shorthand for most of occurrences `@app.env[:uuid]`, so users don't use it. If `uuid` value required, they can get it with getter which we deprecate in favour of `user_id`.

If you sure it's not used I can change it

Reply:
It was a matter of whether `env[:uuid]` might be used outside or not. If not then I can change it although I'm not sure that exactly how you propose is the way to go. I guess I'm going to do it like this:


```ruby
options[:user_id] = options[:uuid] until options[:user_id].nil?
```

Reply:
@kleewho yup, it used internally and user has getter / setters which he can use.
```ruby
options[:user_id] = options[:uuid] until options[:user_id].nil?
```
`until` is for loops and suggested code will hang application.

Proper word `unless`:
```ruby
options[:user_id] = options[:uuid] unless options[:uuid].nil?
```
In this case `unless` mean: _assign passed `uuid` if it is not `nil`_.